libfuse/dir.go

```diff
@@ -408,6 +422,38 @@ func newDir(folder *Folder, node libkbfs.Node) *Dir {
 
 var _ DirInterface = (*Dir)(nil)
 
+// Access implements the fs.NodeAccesser interface for File. See comment for
+// File.Access for more details.
+func (d *Dir) Access(ctx context.Context, r *fuse.AccessRequest) error {
+	if int(r.Uid) != os.Getuid() {
+		// short path: not accessible by anybody other than the logged in user.
+		// This is in case we enable AllowOther in the future.
+		return fuse.EPERM
+	}
+
+	if r.Mask&02 == 0 {
+		// For directory, we only check for the w bit, we can return nil early
+		// here.
+		return nil
+	}
+
+	ei, err := d.folder.fs.config.KBFSOps().Stat(ctx, d.node)
+	if err != nil {
+		if isNoSuchNameError(err) {
+			return fuse.ESTALE
+		}
+		return err
+	}
+	wp, err := d.folder.writePermMode(ctx, &ei, 0)
```

Comment:
Why not call `d.folder.isWriter()` instead?  You just need a boolean, not the full mask, right?  (Same question in the other `Access()` methods.